mm2src/coins/sia/spend_policy.rs

```diff
@@ -0,0 +1,340 @@
+// use super::address::v1_standard_address_from_pubkey;
+use crate::sia::address::Address;
+use crate::sia::blake2b_internal::{public_key_leaf, sigs_required_leaf, standard_unlock_hash, timelock_leaf,
+                                   Accumulator, ED25519_IDENTIFIER};
+use crate::sia::encoding::Encoder;
+use ed25519_dalek::PublicKey;
+use rpc::v1::types::H256;
+
+#[cfg(test)] use std::str::FromStr;
+
+const POLICY_VERSION: u8 = 1u8;
+pub trait Policy {}
+
+#[derive(Debug, Clone)]
+pub enum SpendPolicy {
+    Above(PolicyTypeAbove),
+    After(PolicyTypeAfter),
+    PublicKey(PolicyTypePublicKey),
+    Hash(PolicyTypeHash),
+    Threshold(PolicyTypeThreshold),
+    Opaque(PolicyTypeOpaque),
+    UnlockConditions(PolicyTypeUnlockConditions), // For v1 compatibility
+}
```

Comment:
Some of these variants have unnecessary wrapping, this enum can be written as the following:
```rust
#[derive(Debug, Clone)]
pub enum SpendPolicy {
    Above(u64),
    After(u64),
    PublicKey(PublicKey),
    Hash(H256),
    Threshold(PolicyTypeThreshold),
    Opaque(Address),
    UnlockConditions(UnlockCondition), // For v1 compatibility
}
```